expression/builtin_cast_vec.go

```diff
@@ -608,6 +607,7 @@ func (b *builtinCastDurationAsTimeSig) vecEvalTime(input *chunk.Chunk, result *c
 	ds := buf.GoDurations()
 	times := result.Times()
 	stmtCtx := b.ctx.GetSessionVars().StmtCtx
+	ts, _ := getStmtTimestamp(b.ctx)
```

Comment:
not deal with this error ？

Reply:
> not deal with this error ？

If err is not nil it will return time.Now, this is to be expected

Reply:
But if may return zero time [when error happens](https://github.com/pingcap/tidb/blob/4b50e6745284d2bdc6ac25e248f884f11bb60319/expression/helper.go#L175).

Reply:
> But if may return zero time [when error happens](https://github.com/pingcap/tidb/blob/4b50e6745284d2bdc6ac25e248f884f11bb60319/expression/helper.go#L175).

done.
It will handle error here, if err != nil ts will be set to current time.